crates/rpc-types/src/eth/transaction/request.rs

```diff
@@ -1,51 +1,106 @@
+use alloy_primitives::{Address, Bytes, B256, U256, U64, U8};
 use serde::{Deserialize, Serialize};
 
-/// Represents _all_ transaction requests received from RPC
+/// Represents _all_ transaction requests to/from RPC.
 #[derive(Clone, Debug, PartialEq, Eq, Default, Serialize, Deserialize)]
 #[serde(deny_unknown_fields)]
 #[serde(rename_all = "camelCase")]
 pub struct TransactionRequest {
-    /// from address
+    /// The address of the transaction author.
     pub from: Option<Address>,
-    /// to address
+    /// The destination address of the transaction.
     pub to: Option<Address>,
-    /// legacy, gas Price
+    /// The legacy gas price.
     #[serde(default)]
-    pub gas_price: Option<U128>,
-    /// max base fee per gas sender is willing to pay
+    pub gas_price: Option<U256>,
+    /// The max base fee per gas the sender is willing to pay.
     #[serde(default)]
-    pub max_fee_per_gas: Option<U128>,
-    /// miner tip
+    pub max_fee_per_gas: Option<U256>,
+    /// The max priority fee per gas the sender is willing to pay, also called the miner tip.
     #[serde(default)]
-    pub max_priority_fee_per_gas: Option<U128>,
-    /// gas
+    pub max_priority_fee_per_gas: Option<U256>,
+    /// The max fee per blob gas for EIP-4844 blob transactions.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub max_fee_per_blob_gas: Option<U256>,
+    /// The gas limit for the transaction.
     pub gas: Option<U256>,
-    /// value of th tx in wei
+    /// The value transferred in the transaction, in wei.
     pub value: Option<U256>,
-    /// Any additional data sent
-    #[serde(alias = "input")]
-    pub data: Option<Bytes>,
-    /// Transaction nonce
+    /// Transaction data.
+    #[serde(default, flatten)]
+    pub input: TransactionInput,
+    /// The nonce of the transaction.
     pub nonce: Option<U64>,
-    /// warm storage access pre-payment
+    /// The chain ID for the transaction.
+    pub chain_id: Option<U64>,
+    /// An EIP-2930 access list, which lowers cost for accessing accounts and storages in the list. See [EIP-2930](https://eips.ethereum.org/EIPS/eip-2930) for more information.
     #[serde(default)]
     pub access_list: Option<AccessList>,
-    /// EIP-2718 type
+    /// The EIP-2718 transaction type. See [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) for more information.
     #[serde(rename = "type")]
     pub transaction_type: Option<U8>,
-    /// sidecar for EIP-4844 transactions
+    /// Blob versioned hashes for EIP-4844 transactions.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub blob_versioned_hashes: Option<Vec<B256>>,
+    /// Blob sidecar for EIP-4844 transactions.
     pub sidecar: Option<BlobTransactionSidecar>,
     /// Support for arbitrary additional fields.
     #[serde(flatten)]
     pub other: OtherFields,
 }
 
+impl Hash for TransactionRequest {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.from.hash(state);
```

Comment:
why can't we derive this?

Reply:
`OtherFields` contains `serde_json::Value` which is not `Hash`